kotlin/internal/jvm/compile.bzl

```diff
@@ -346,8 +351,12 @@ def _run_kt_builder_action(
     for f, path in outputs.items():
         args.add("--" + f, path)
 
-    args.add_all("--kotlin_passthrough_flags", _kotlinc_options_provider_to_flags(toolchains.kt.kotlinc_options))
-    args.add_all("--javacopts", _javac_options_provider_to_flags(toolchains.kt.javac_options))
+    # Unwrap kotlinc_opts/javac_options options or default to the ones being provided by the toolchain
+    kotlinc_options = ctx.attr.kotlinc_opts[_KotlincOptions] if ctx.attr.kotlinc_opts else toolchains.kt.kotlinc_options
```

Comment:
We should make an explicit decision about how the merging strategy should go. I'm not opposed to rule prioritized over toolchain, but it needs documentation. And possibly an RFC. @cgruber, your thoughts?

Reply:
We looked into merging the opts before passing them to the builder but landed on just prioritizing the rule over toolchain because it makes opts consumption more explicit and has room for hidden side effects.

Reply:
I consider prioritizing the rule over toolchain a merging strategy, and would need to be documented as such. It is the most intuitive approach.